stout/flags.cc

```diff
@@ -18,63 +18,185 @@ namespace stout::flags {
+                        << "'" << name << "' for message '"
+                        << message->GetTypeName() << "'"
+                        << std::endl;
+              std::exit(1);
+            }
+          }
+
+          // Create nested parser and store it to the 'nested_parsers_'.
+          auto [_, inserted] = nested_parsers_.emplace(
+              field,
+              std::make_unique<Parser>(
+                  Parser::Builder(
+                      message->GetReflection()->MutableMessage(message, field),
+                      true)
+                      .Build()));
 
-    if (flag.help().empty()) {
-      std::cerr
-          << "Missing flag 'help' for field '" << field->full_name() << "'"
-          << std::endl;
-      std::exit(1);
-    }
+          CHECK(inserted);
+        }
+      }
+    } else {
+      if (field->options().HasExtension(stout::v1::subcommand)) {
+        std::cerr << "stout.v1.subcommand option should be annotated"
+                  << " on fields that are only inside 'oneof subcommand'"
+                  << std::endl;
+        exit(1);
+      }
 
-    for (const auto& name : flag.names()) {
-      AddOrExit(name, field, message);
-    }
+      const auto& flag = field->options().GetExtension(stout::v1::flag);
+
+      if (flag.names().empty()) {
+        std::cerr
+            << "Missing at least one flag name in 'names' for field '"
+            << field->full_name() << "'"
+            << std::endl;
+        std::exit(1);
+      }
+
+      if (flag.help().empty()) {
+        std::cerr
+            << "Missing flag 'help' for field '"
+            << field->full_name() << "'"
+            << std::endl;
+        std::exit(1);
+      }
+
+      for (const auto& name : flag.names()) {
+        auto [_, inserted] = fields_.emplace(name, field);
+
+        if (!inserted) {
+          std::cerr << "Encountered duplicate flag name "
+                    << "'" << name << "' for message '"
+                    << message->GetTypeName() << "'"
+                    << std::endl;
+          std::exit(1);
+        }
+      }
+
+      for (const auto& name : flag.deprecated_names()) {
+        auto [_, inserted] = fields_.emplace(name, field);
 
-    for (const auto& name : flag.deprecated_names()) {
-      AddOrExit(name, field, message);
+        if (!inserted) {
+          std::cerr << "Encountered duplicate (deprecated) flag name "
+                    << "'" << name << "' for message '"
+                    << message->GetTypeName() << "'"
+                    << std::endl;
+          std::exit(1);
+        }
+      }
     }
   }
 }
 
 ////////////////////////////////////////////////////////////////////////
 
-void Parser::AddOrExit(
-    const std::string& name,
-    const google::protobuf::FieldDescriptor* field,
-    google::protobuf::Message* message) {
-  auto [_, inserted] = fields_.emplace(name, field);
+Parser* Parser::TryLookupParserForSubcommand(const std::string& name) {
```

Comment:
```suggestion
Parser* Parser::TryLookupNestedParserForSubcommand(const std::string& name) {
```

Reply:
Happy to have you add this in a subsequent PR.